### DISCUSSION TITLE
Add a way of installing fastfetch on the Librem 5 phone

### DISCUSSION THREAD

**Discussion by CocolinoFan** [fastfetch-cli/fastfetch#1130] [Ideas]
Hello,

Is it possible to install `fastfetch` on my Librem 5 phone running PureOS Linux? The old `neofetch` does not have an ASCII art for the OS.
I have tried: adding `ppa:zhangsongcui3371/fastfetch` and installing `linuxbrew` but no success. 
I have no tried yet to build from source but, that would not be very useful to me anyway as updating would be a nightmare.

**Comment by CarterLi** [comment]
> adding ppa:zhangsongcui3371/fastfetch

What error did it report?

**Reply by CocolinoFan** [reply]
`ppa:zhangsongcui3371/fastfetch` says: 
"Some packages could not be installed. This may mean that you have requested an impossible situation or if you are using the unstable distribution that some required packages have not yet been created or been moved out of lncoming.
The following information may help to resolve the situation:
The following packages have unmet  dependencies:
fastfetch: Depends: libc6 (>= 2.34) but 2.31-13+deb11u10 is to be installed
E: Unable to correct problems. you have held broken packages.
"
`linuxbrew` is not available for arm processors.

**Comment by CarterLi** [comment]
Please try this: https://github.com/CarterLi/fastfetch/actions/runs/10145571439/artifacts/1751081345

**Reply by CocolinoFan** [reply]
Thank you sir, the .deb package worked great. 
Will it now get automatic updates?
It has the same artwork as `neofetch` :<  
![100% screen resolution](https://github.com/user-attachments/assets/12e90d58-d560-402c-9074-dedef98cb9d4)
![200% zoom - the default state of the phone](https://github.com/user-attachments/assets/5bf7b999-2d37-4901-abd9-8c58922211a3)

P.S. Excuse the spacing between letters, I did something funky with the fonts that I am trying to undo.

**Reply by CarterLi** [reply]
The advantage of fastfetch is that fastfetch is maintained. I will update the logo if you can provide one.

**Reply by CocolinoFan** [reply]
I've been doing some thinking and `fastfetch`'s ASCII art is good. Is the PureOS logo itself that's just really basic 😅
Now I just need to figure out why, when the phone is in horizontal orientation and the screen is at 200% zoom (that's the default, 2nd image) the logo is not in line with the text; and make a pull request for it :) (might have something to do with my wired letter-spaceing issue)
Also, how should my fellow Librem 5 users install this great software? I would need to tell my distribution's maintainers to make a .deb files themselves for every update and add it to their repositories?

**Reply by CarterLi** [reply]
PureOS is based on Debian. As fastfetch is accepted into debian repo, you will likely be able to install fastfetch directly by `apt`, in the near future.

**Comment by CarterLi** [comment]
It's the first time I see someone running fastfetch on linux phone. Can you upload the output of `fastfetch -c all --format json` for debugging?

**Reply by CocolinoFan** [reply]
Yes sir :3
[debugLibrem5.json](https://github.com/user-attachments/files/16415713/debugLibrem5.json)
